### PR TITLE
For #22637 - Add new color tokens provided by the mobile styles to FirefoxTheme

### LINKED ISSUE #22637: Add new color tokens provided by the design systems to FirefoxTheme
We have some new design color tokens that are part of the mobile design system that we can add to FirefoxTheme that can be used in our @Composables.

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #22637, #22654

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by rocketsroger — APPROVED** [review]
🚢

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt [hunk 1]
```diff
@@ -202,7 +202,10 @@ private fun RecentSearchGroupItem(
                         painter = painterResource(id = R.drawable.ic_all_tabs),
                         modifier = Modifier.size(18.dp),
                         contentDescription = null,
-                        tint = FirefoxTheme.colors.textSecondary
+                        tint = when (isSystemInDarkTheme()) {
+                            true -> FirefoxTheme.colors.textPrimary
+                            false -> FirefoxTheme.colors.textSecondary
+                        }
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:208]: This color determination could make sense as a method on `FirefoxTheme` if it's going to be re-used regularly, IMO

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:208]: Good point, I am hoping it won't be and that we will align with textSecondary, but that requires some conversation with UX to inquire why this is different for the dark theme and how this is not consistent everywhere.

**Comment by gabrielluong** [comment]
`[task 2021-12-03T00:10:35.513Z] FAILURE: Build failed with an exception.
[task 2021-12-03T00:10:35.513Z] 
[task 2021-12-03T00:10:35.513Z] * What went wrong:
[task 2021-12-03T00:10:35.513Z] A problem occurred configuring root project 'vcs'.
[task 2021-12-03T00:10:35.513Z] > Could not resolve all artifacts for configuration ':classpath'.
[task 2021-12-03T00:10:35.513Z]    > Could not resolve org.mozilla.telemetry:glean-gradle-plugin:42.0.1.
[task 2021-12-03T00:10:35.513Z]      Required by:
[task 2021-12-03T00:10:35.513Z]          project : > org.mozilla.components:tooling-glean-gradle:96.0.20211202171215
[task 2021-12-03T00:10:35.513Z]       > Could not resolve org.mozilla.telemetry:glean-gradle-plugin:42.0.1.
[task 2021-12-03T00:10:35.513Z]          > Could not get resource 'https://maven.mozilla.org/maven2/org/mozilla/telemetry/glean-gradle-plugin/42.0.1/glean-gradle-plugin-42.0.1.module'.
[task 2021-12-03T00:10:35.514Z]             > Could not GET 'https://maven.mozilla.org/maven2/org/mozilla/telemetry/glean-gradle-plugin/42.0.1/glean-gradle-plugin-42.0.1.module'. Received status code 503 from server: Service Unavailable
[task 2021-12-03T00:10:35.514Z] 
[task 2021-12-03T00:10:35.514Z] * Try:
[task 2021-12-03T00:10:35.514Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[task 2021-12-03T00:10:35.514Z] 
[task 2021-12-03T00:10:35.514Z] * Get more hel`

**Pull request merged**